plugin/forward/persistent.go

```diff
@@ -26,17 +28,21 @@ type transport struct {
 	yield chan *dns.Conn
 	ret   chan *dns.Conn
 	stop  chan bool
+
+	avgDialTime int64 // kind of average time of dial time
```

Comment:
@rdrozhdzh : this average is per Forward.Proxy (each Forward.Proxy maintain its own Transport object). I mean each computation is dedicated for one upstream. The first dialTimeout is the default, and then, if the line fall because expireTime or else, the next retry Dial will use an average (sort-of) of the last real time needed to Dial.

Reply:
oh, sorry, you are right.